overlay.py

```diff
@@ -50,9 +59,10 @@ def publish_recording_status(self, message: str) -> None:
 	def connect(self, ip="192.168.100.100", port=1883):
 		self.client.connect_async(ip, port, 60)
 
-		# Raspberry Pis run ARM, PCs run x86_64
-		backend_name = "picamera" if machine() == "armv71" else "opencv"
-		with BackendFactory.create(backend_name, self.width, self.height, self.publish_recording_status) as self.backend:
+		with BackendFactory.create(self.backend_name, self.width, self.height, self.publish_recording_status) as self.backend:
+
+			if self.backend_name == "opencv_static_image":
+				self.backend.set_background(self.bg_path)
```

Comment:
I'm not totally happy with how I've set this up to pass the background path... Any suggestions are very much appreciated!

Reply:
I'm not happy with this either but I actually don't know how to make this part a bit more elegant. It's also a bit annoying that we have to pass down the `bg` parameter to all the overlays as well. This is a problem cos not all `Overlay`'s will have a background.

One way we could possibly make this a bit more cleaner is if we pass in a `video_feed` argument into the factory which either takes in `cv2.VideoCapture(default_camera_index)`, `image_filepath` or `PiCamera(resolution=(self.width, self.height))`. That way we should be able to set an image using opencv or picamera? Might have to look into upgrading our Factory into an Abstract Factory (https://refactoring.guru/design-patterns/abstract-factory) so basically a factory that produces other factories. In this case, we would have a `PiCameraFactory` and a `OpenCVFactory` which both produce a `StaticImageProduct` and a `VideoProduct` (names are for example only btw).

This is probably work for another PR though, happy for ideas though.

Reply:
Hmm... There are things that irk me about both of the solutions you suggest:
- If we pass a `video_feed` argument, it means we'll have to import both OpenCV and picamera into `overlay.py`, when really that file shouldn't have to worry about those libraries. Also, I'm not a huge fan of passing such random objects into the same argument.
- I don't think the abstract factory design would really work here, because (for example), we would not be able to load a static image using the picamera library, meaning that the two factories produce a different subset of products.

Reply:
Ohhh okay, I would have thought that the `picamera` library would allow static images to be loaded. But in that case, that's fair enough. As long as this if statement does not continue to grow with the number of factories then we don't need to change the current factories.